streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -363,6 +362,98 @@ public synchronized final TopologyBuilder addSource(String name, Deserializer ke
     }
 
     /**
+     * Add a global store to the topology. This store can be accessed by any of the Processors
+     * in the topology.
+     *
+     * @param store  the instance of {@link StateStore}
+     * @return this builder instance so methods can be chained together; never null
+     */
+    public synchronized TopologyBuilder addGlobalStore(final StateStore store) {
+        Objects.requireNonNull(store, "store must not be null");
+        if (stateFactories.containsKey(store.name()) || globalStateStores.containsKey(store.name())) {
+            throw new TopologyBuilderException("StateStore " + store.name() + " is already added.");
+        }
+        globalStateStores.put(store.name(), store);
+        return this;
+    }
+
+    /**
+     * Adds a global {@link StateStore} to the topology. The {@link StateStore} sources its data
+     * from all partitions of the provided input topic. There will be exactly one instance of this
+     * {@link StateStore} per Kafka Streams instance.
+     * <p>
+     * A {@link SourceNode} with the provided sourceName will be added to consume the data arriving
+     * from the partitions of the input topic.
+     * <p>
+     * The provided {@link ProcessorSupplier} will be used to create an {@link ProcessorNode} that will
+     * receive all records forwarded from the {@link SourceNode}. This
+     * {@link ProcessorNode} should be used to keep the {@link StateStore} up-to-date.
+     *
+     * @param store                 the instance of {@link StateStore}
+     * @param sourceName            name of the {@link SourceNode} that will be automatically added
+     * @param keyDeserializer       the {@link Deserializer} to deserialize keys with
+     * @param valueDeserializer     the {@link Deserializer} to deserialize values with
+     * @param topic                 the topic to source the data from
+     * @param processorName         the name of the {@link ProcessorSupplier}
+     * @param stateUpdateSupplier   the instance of {@link ProcessorSupplier}
+     * @return
+     */
+    public synchronized TopologyBuilder addGlobalStore(final StateStore store,
```

Comment:
This is in the KIP.
Anyway, this method has been added so that the joins generated by `GlobalKTables` can be queried etc. As they are just views on top of other `GlobalKTables` they don't have their own source as such.
Also, from a users point of view - why shouldn't they be able to add a global store of whatever type they like? They might have a pre-populated table or something that they'd like available in all of there Processors.